src/queue/memoryQueue.ts

```diff
@@ -0,0 +1,105 @@
+import { QueueId, QueueInterface } from "./queue";
+import { DSNPMessage, DSNPType } from "../messages/messages";
+
+const InvalidId = new Error("The Queue Id provided is malformed.");
+const IdDoesNotExist = new Error("No message matching the given ID exists.");
+
+interface QueueItem {
+  id: number;
+  message: DSNPMessage;
+}
+
+/**
+ * MemoryQueue provides a simple, in-memory store for DSNP messages to be
+ * announced. This adapter is the default if no other is provided in
+ * configuration.
+ */
+export default class MemoryQueue implements QueueInterface {
+  queues: Record<string, QueueItem[]>;
+  nextId: number;
+
+  constructor() {
+    this.queues = {};
+    this.nextId = 0;
+  }
+
+  /**
+   * enqueue() implements the enqueue function of the QueueInterface. It takes a
+   * DSNP message, adds it to an in-memory store and returns an id which can be
+   * used to reverse this action, if needed.
+   *
+   * @param message The DSNP message to queue up for the next batch
+   * @returns       A queue id to be used to remove the message if needed
+   */
+  async enqueue(type: DSNPType, message: DSNPMessage): Promise<QueueId> {
```

Comment:
Doesn't a DSNPMessage contain the type?

Reply:
It does, but I was thinking pulling the type out would make implementation easier for future storage adapters like Redis. I'll change it.